### ISSUE TITLE
ejbCreate not called when mpMetric feature is on, and when metadata-complete is true

### ISSUE THREAD

**Issue by tschurins** [OpenLiberty/open-liberty#11996]
**Describe the bug**
We have an application with EJBs and MDBs still in the 2.1 form (with interface, and home, and bean and ejb-jar definition; no annotation). The EJBs and MDBs have some initialisation done in the ejbCreate method. For performance reasons, we also set the metadata-complete flag to true in the ejb-jar definition.
So far, so good, everything is working perfectly.

Now we want to enable the metrics feature by adding in the list of feature the mpMetrics (2.0 for example). This should not have any impact on the code or the behaviour of the application.
Unfortunately, this is not the case. The ejbCreate methods of the EJBs and MDBs is not called anymore.

**Steps to Reproduce**
I have attached a maven project creating a ear with one EJB. The ear has also a servlet to showcase the problem. See readme.md

[ejbCreateEAR.zip](https://github.com/OpenLiberty/open-liberty/files/4558989/ejbCreateEAR.zip)

**Expected behavior**
The ejbCreate method should be called regardless of the presence of the mpMetrics feature.

**Diagnostic information:**
 - OpenLiberty Version: 19.0.0.9, also tested on 20.0.0.3
 - Docker image from https://hub.docker.com/_/open-liberty

**Additional context**
I took the opportunity to dig into the code:
In `com.ibm.ws.metadata.ejb.EJBMDOrchestrator#initializeInterceptorMD`, when mpMetrics is enabled, there are InterceptorClasses (added by the mpMetrics feature); in this case the callback kind is CallbackKind.InvocationContext, and the lifecycle methods are based on BeanInterceptorProxyMap which is empty (it comes from `com.ibm.ws.metadata.ejb.InterceptorMetaDataFactory#processBeanInterceptors` which introspects the class annotations and lifecycle methods (ejbCreate &c.) only if metadata-complete is false).

On the other hand, when mpMetrics is not present, there are no InterceptorClasses there, so the callback kind is CallbackKind.MessageDrivenBean, and the EjbCreateMethod is set to ejbCreate.

**Comment by tkburroughs** [comment]
Agree; this is a bug in `com.ibm.ws.metadata.ejb.InterceptorMetaDataFactory#processBeanInterceptors`.

With `metadata.-omplete="true"` the code should not be looking for annotations, however an `ejbCreate()` method is special and required to be a post construct interceptor regardless of annotations.  There is code that already looks for it without an annotation, but right now it is only performed it `metadata-complete="false"`.

A workaround would be to specifically designate `ejbCreate` as the post-construct method in ejb-jar.xml:

```
            <post-construct>
               <lifecycle-callback-method>initUserTransaction</lifecycle-callback-method>
            </post-construct>
```

**Comment by tschurins** [comment]
Yes, this workaround is indeed working.
